checkov/common/runners/object_runner.py

```diff
@@ -100,6 +100,9 @@ def run(
         for file_path in self.definitions.keys():
             self.pbar.set_additional_data({'Current File Scanned': os.path.relpath(file_path, root_folder)})
             skipped_checks = collect_suppressions_for_context(self.definitions_raw[file_path])
+
+            if registry.report_type == CheckType.GITLAB_CI:
+                registry.set_definitions_raw(self.definitions_raw[file_path])
```

Comment:
any reason, why limit it to just GitLab CI?

Reply:
I wasn't sure if that's needed for all frameworks, I added it because some gitlab ci check names were missing the resource lines and code blocks and I needed the definitions to resolve them.
I think we should gradually apply it to more frameworks as we find issues with their resource ids, WDYT?

Reply:
gotcha, gotcha. let's leave it for now as it is. if we need it for something else, then lets add it for all checks